### PR TITLE
[16.0][ADD] purchase_only_by_packaging

### PR DESCRIPTION
**Pull request by santostelmo**
This module feature was extracted from the original [sale-workflow/sale_by_packaging](https://github.com/oca/sale-workflow/tree/14.0/sale_by_packaging) module.

This module supports 2 features based on sale_by_packaging v14:

purchase only by packaging: On product template model, this checkbox restricts
purchases of these products if no packaging is selected on the purchase order line.
If no packaging is selected, it will either be auto-assigned if the quantity
on the purchase order line matches a packaging quantity or an error will be raised.

Force purchase quantity (on the packaging): force rounds up the quantity during
creation/modification of the purchase order line with the factor set on the packaging.

Depend on:
- https://github.com/OCA/product-attribute/pull/1436

Base on:
- https://github.com/OCA/sale-workflow/pull/2623

### CODE REVIEW

File: purchase_only_by_packaging/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+{
+    "name": "Purchase Only By Packaging",
+    "summary": "Manage purchase of packaging",
+    "version": "16.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Warehouse Management",
+    "website": "https://github.com/OCA/purchase-workflow",
+    "author": "Camptocamp, Odoo Community Association (OCA)",
```
Comment (jbaudoux) [inline purchase_only_by_packaging/__manifest__.py]:
```suggestion
    "author": "Camptocamp, BCIM, Odoo Community Association (OCA)",
```

**Review by jbaudoux — APPROVED** [review]
LGTM

**Comment by jbaudoux** [comment]
Based on https://github.com/OCA/sale-workflow/pull/2623 instead of https://github.com/OCA/sale-workflow/pull/2570.

I don't know how you put the links in the description but we don't see their status.

**Review by cyrilmanuel — APPROVED** [review]

**Comment by simahawk** [comment]
@santostelmo tests are :red_circle:

**Comment by santostelmo** [comment]
> @santostelmo tests are 🔴

@simahawk test is falling as we depend on https://github.com/OCA/product-attribute/pull/1436 .

**Comment by simahawk** [comment]
dep merged

**Comment by santostelmo** [comment]
@simahawk  running tests we still have:
`ERROR: No matching distribution found for odoo-addon-product-packaging-level-purchasable<16.1dev,>=16.0dev`
Not sure since  `product-packaging-level-purchasable`is already merded

**Comment by simahawk** [comment]
We had a major failure on OCA services last weeks and the wheelhouse was probably down when the PR got merged.
I'll do a small change on that module and try to fix the release.

**Comment by santostelmo** [comment]
Rebased

**Pull request merged**